### PR TITLE
Document CI structure

### PR DESCRIPTION
**Pull request by martinpitt**
For reviewing I suggest to look at the rendered result here: https://github.com/martinpitt/kickstart-tests/tree/ci-docs#chapter-7-continuous-integration-structure

### CODE REVIEW

**Comment by martinpitt** [comment]
Asking for many reviews to make sure that this all makes sense to you. Does this have any implicit assumptions or knowledge that are not described?

File: README.rst [hunk 1]
```diff
@@ -309,7 +309,75 @@ reason is that using the inject method the network devices are not initialized
...
+workflow files and the runner container.
+
+The results can be viewed on the `GitHub Daily run workflows page`_. Each run
+has an artifact attached with the detailed log files. This is currently not
+very comfortable, and we are actively looking for a better solution how to
+publish the test result history.
+
+These tests are expected to succeed normally. On failures, rhinstaller
+maintainers get a "failed workflow" notification email and should investigate
+the cause.
```
Comment (rvykydal) [inline README.rst:341]: Should we maybe mention our flakes handling here ? At least give some pointer ? Or is it too much of detail / configuration for this place ?

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

**Review by jkonecny12 — APPROVED** [review]
Looks great to me!

File: README.rst (hunk 1 above)
Comment (martinpitt) [reply README.rst:341]: Good point, thanks! Updated accordingly, PTAL?

Comment (rvykydal) [reply README.rst:341]: Hm, I can't find the update.

Comment (martinpitt) [reply README.rst:341]: Bah, git pilot error, sorry. [for reals now!](https://github.com/rhinstaller/kickstart-tests/compare/3618af2107d50088cb77488e87f1d029925a21c8..c77863fcdf02db17541956819ae114ac2ab18165)

**Review by rvykydal — APPROVED** [review]
Looks great to me.

**Pull request merged**